Module.php

```diff
@@ -31,12 +32,18 @@ public function getTitle()
     public function getHtmlCode()
     {
         $htmlCode = $this->settings->get('htmlCode');
+
         if (empty($htmlCode)) {
             return '';
         }
-        return $htmlCode;
+
+        $nonce = Html::nonce();
+        $htmlNonce = preg_replace('/<script([^>]*)>/', '<script$1 nonce="' . $nonce . '">', $htmlCode);
```

Comment:
If this doesn't work as needed then `$htmlNonce = str_replace('nonce={{nonce}}', $nonce, $htmlCode);` without a value should.

Reply:
Didn't test it but I would expect: 

```suggestion
        $htmlNonce = preg_replace('/<script([^>]*)>/', '<script$1 ' . $nonce . '>', $htmlCode);
```